perf-tool/src/server.cpp

```diff
@@ -158,24 +159,56 @@ void Server::handleServer()
             }
         }
 
-        // Receiving and sending messages from/to clients
+        /* Receiving messages to clients */
         for (int i = 0; i < activeNetworkClients; i++)
         {
-            bzero(buffer, 512);
-            command = networkClients[i]->handlePoll(buffer);
+            command = networkClients[i]->handlePoll();
             if (!command.empty())
             {
                 handleClientCommand(command, "Client");
             }
         }
+
+        /* Checks if it is the time for any delayed command to be fired */
+        if (!delayedCommands.empty()) {
+            auto currentClockTime = std::chrono::system_clock::now();
+            std::time_t currentTime = std::chrono::system_clock::to_time_t(currentClockTime);
+
+            for (int i = 0; i < delayedCommands.size(); i++)
+            {
+                if (delayedCommands[i].delayTill <= currentTime) {
+                    agentCommand(delayedCommands[i].command);
+                    delayedCommands.erase(delayedCommands.begin()+i);
```

Comment:
Deleting while traversing is tricky. Once you delete an element, all the others are moved towards front to keep the vector compact, thus you should not advance the pointer.
https://stackoverflow.com/questions/9927163/erase-element-in-vector-while-iterating-the-same-vector